modules/lwjgl/core/src/templates/kotlin/core/windows/templates/Crypt32.kt

```diff
@@ -0,0 +1,175 @@
+        "CRYPTPROTECT_PROMPT_ON_PROTECT"..0x2
+    )
+
+    IntConstant(
+        "The block size for data encrypted via #CryptProtectMemory().",
+        "CRYPTPROTECTMEMORY_BLOCK_SIZE".."16"
+    )
+
+    SaveLastError..BOOL(
+        "CryptProtectData",
+        """
+        The {@code CryptProtectData} function performs encryption on the data in a ##DATA_BLOB structure. Typically, only a user with the same logon credential
+        as the user who encrypted the data can decrypt the data. In addition, the encryption and decryption usually must be done on the same computer.
+        """,
+
+        DATA_BLOB.p("pDataIn", "a pointer to a ##DATA_BLOB structure that contains the plaintext to be encrypted"),
+        nullable..LPCWSTR(
+            "szDataDescr",
+            "a string with a readable description of the data to be encrypted. This description string is included with the encrypted data."
+        ),
+        nullable..DATA_BLOB.p(
+            "pOptionalEntropy",
+            """
+            A pointer to a ##DATA_BLOB structure that contains a password or other additional entropy used to encrypt the data. The {@code DATA_BLOB} structure
+            used in the encryption phase must also be used in the decryption phase. This parameter can be set to #NULL for no additional entropy.
+            """
+        ),
+        Unsafe..nullable..void.p("pvReserved", "reserved for future use and must be set to #NULL"),
+        nullable..CRYPTPROTECT_PROMPTSTRUCT.p(
+            "pPromptStruct",
+            """
+            a pointer to a ##CRYPTPROTECT_PROMPTSTRUCT structure that provides information about where and when prompts are to be displayed and what the content
+            of those prompts should be. This parameter can be set to #NULL in both the encryption and decryption phases.
+            """
+        ),
+        DWORD(
+            "dwFlags",
+            "this parameter can be",
+            "#CRYPTPROTECT_LOCAL_MACHINE, #CRYPTPROTECT_UI_FORBIDDEN, #CRYPTPROTECT_AUDIT",
+            LinkMode.BITFIELD
+        ),
+        DATA_BLOB.p(
+            "pDataOut",
+            """
+            a pointer to a ##DATA_BLOB structure that receives the encrypted data. When you have finished using the {@code DATA_BLOB} structure, free its
+            {@code pbData} member by calling the #LocalFree() function.
+            """
+        )
+    )
+
+    SaveLastError..BOOL(
+        "CryptProtectMemory",
+        """
+        The {@code CryptProtectMemory} function encrypts memory to prevent others from viewing sensitive information in your process. For example, use the
+        {@code CryptProtectMemory} function to encrypt memory that contains a password. Encrypting the password prevents others from viewing it when the process
+        is paged out to the swap file. Otherwise, the password is in plaintext and viewable by others.
+        """,
+
+        void.p("pDataIn", "a pointer to the block of memory to encrypt"),
```

Comment:
`pDataIn` is actually defined as `LPVOID` ([ref](https://learn.microsoft.com/en-us/windows/win32/api/dpapi/nf-dpapi-cryptprotectmemory)). However, this is mapped to a raw pointer instead of a buffer and does not also not allow us to autosize the `cbDataIn` parameter. This, I chose `void.p` instead. Is this fine, or should I change this to align with the definition?

Reply:
Changed to `typedef(void.p, "LPVOID")("pDataIn", "a pointer to the block of memory to encrypt")`. Functionally the same, but changes the annotation to `@NativeType("LPVOID")`.